File: python/nt_solutions/sparsity_6_l1_recovery/exo6.py
g = lambda C,I: np.sum(C[:,I], 1)

plt.figure(figsize=(8,5))
dlist = np.arange(1,N/20)
criter = np.zeros([len(dlist),3])

for i in range(len(dlist)):
    s = twosparse(dlist[i])
    I = (supp(s))
    criter[i,:] = [F(Phi, s), erc(Phi,I), werc(Phi,I)]
    
criter[criter < 0] = np.float("inf")

plt.plot(dlist, criter, linewidth=2)
plt.plot(dlist, dlist*0 + 1, 'k--', linewidth=2)
plt.xlim(1,np.max(dlist))
plt.ylim(np.min(criter), np.max(criter))
plt.xlabel("d")
plt.legend(["F", "ERC", "w-ERC"])
           
plt.show()

File: r/nt_solutions/sparsity_6_l1_recovery/exo6.R
g = function(C,I){apply(C[,I], 1, sum)}

dlist = c(1:(N/20))
criter = matrix(0, length(dlist),3)

for (i in 1:length(dlist))
{
    s = twosparse(dlist[i])
    I = supp(s)
    criter[i,] = c(F(Phi, s), erc(Phi,I), werc(Phi,I))
}
    
criter[criter < 0] = NA

options(repr.plot.width=7, repr.plot.height=5)

matplot(dlist, criter, xlim=c(1, max(dlist)), ylim=c(min(criter), max(criter)), type="l", col=c(4, 3, 2),
        lty=1, xlab='', ylab='')
legend("right", legend=c('F', 'ERC', 'w-ERC'), 
       col=c(4, 3, 2), pch="-")
par(new=TRUE)
plot(dlist, dlist * 0 + 1, xlim=c(1, max(dlist)), ylim=c(min(criter), max(criter)), type="l", lty=2, ylab='', xlab='')
           